### PR TITLE
refactor: remove global network usage from petnames

### PR DESCRIPTION
**Pull request by matthewwalsh0**
## **Description**

Remove all usage of the global network from the `Name` component and related display name hooks.

Specifically:

- Remove usages of the `selectChainId` selector.
- Require `variation` property in `Name` component and `useDisplayName` hook.
- Add linting rule to prevent future usage of any global network selectors.
- Add `imageUri` property to `Identicon` component.
- Add `chainId` property to `Address` component.
- Rename `useTokenListEntry` hook to `useERC20Tokens`.
- Add `selectERC20TokensByChain` selector.
- Remove `useTokenList` hook.

## **Related issues**

Fixes: [#1987](https://github.com/MetaMask/mobile-planning/issues/1987)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -73,6 +73,33 @@ module.exports = {
...
+              'selectProviderConfig',
+              'selectProviderType',
+              'selectRpcUrl',
+              'selectSelectedNetworkClientId',
+              'selectTicker'
+            ]
+              .map((method) => `(${method})`)
+              .join('|')}/]`,
+            message: 'Avoid using global network selectors in confirmations',
+          },
```
Comment (jpuri) [inline .eslintrc.js:99]: Should this rather be warning if possible, as use may need to access these selectors.

**Review by jpuri — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

File: app/components/UI/Identicon/index.tsx [hunk 2]
```diff
@@ -42,17 +42,14 @@ const Identicon: React.FC<IdenticonProps> = ({
...
+  imageUri,
 }) => {
   const { colors } = useTheme();
-  const tokenListIcon = useTokenListEntry(
-    address || '',
-    NameType.EthereumAddress,
-  )?.iconUrl;
 
-  if (!address) return null;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (tommasini) [inline app/components/UI/Identicon/index.tsx:49]:
```suggestion

```

**Review by tommasini — APPROVED** [review]

File: app/components/UI/SimulationDetails/useSimulationMetrics.test.ts [hunk 3]
```diff
@@ -22,10 +23,13 @@ import {
   useSimulationMetrics,
 } from './useSimulationMetrics';
 import useLoadingTime from './useLoadingTime';
+import { selectChainId } from '../../../selectors/networkController';
 
 jest.mock('react-redux', () => ({
   ...jest.requireActual('react-redux'),
   useDispatch: jest.fn(),
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
[the remark this answers is not in the document]
Comment (matthewwalsh0) [reply app/components/UI/SimulationDetails/useSimulationMetrics.test.ts:31]: This selector and mock will be removed in the next PR to remove global network from simulations 👍

File: app/components/UI/Identicon/index.tsx (hunk 2 above)
Comment (matthewwalsh0) [reply app/components/UI/Identicon/index.tsx:49]: Will remove this in the next simulations global network PR 👍

**Pull request merged**